nix/keys.nix

```diff
@@ -133,6 +155,8 @@ in
               let
                 # FIXME: The key file should be marked as private once
                 # https://github.com/NixOS/nix/issues/8 is fixed.
+                #
+                # FIXME: implement value.filename for this as well
```

Comment:
If only `filename` is set and `deployment.storeKeysOnMachine` is set to true, this will fail since `value.text` one line below will be `null`.

We have to add another assertion to say that `deployment.storeKeysOnMachine && filename` is not supported.